src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -81,7 +82,7 @@ var ReactDOMInput = {
       onChange: inst._wrapperState.onChange,
     });
 
-    return nativeProps;
+    return DisabledInputUtils.getNativeProps(inst, nativeProps);
```

Comment:
Good point. Looking at this again, I've got a new thought regarding `assign`. I'll post that in a second.